### PR TITLE
Add test for add single email function

### PR DESCRIPTION
**Pull request by dominikhajduk**
@php-refactor please look on this and review.

### CODE REVIEW

File: tests/functional/AddMailTest.php [hunk 1]
```diff
@@ -20,16 +20,30 @@ public function setUpPage()
...
+
         $button = $this->byName('add_mail');
 
         $button->click();
 
-        $this->assertContains('Niepoprawne mail!', $this->source());
+        $this->assertContains($expectedMessage, $this->source());
+    }
+
+    public function emailProvider()
```
Comment (partikus) [inline tests/functional/AddMailTest.php]: @dominikhajduk ``emailScenarios`` ?

Comment (dominikhajduk) [inline tests/functional/AddMailTest.php]: @partikus you mean change method name?

Comment (partikus) [inline tests/functional/AddMailTest.php]: yes, exactly I meant changing the name

**Comment by dominikhajduk** [comment]
@partikus changed

**Comment by partikus** [comment]
:+1:

**Comment by dominikhajduk** [comment]
@php-refactor Add database clean up. Without that only first test will be correct. I can not find nice way to use both database and Selenium extension in single test.

**Comment by partikus** [comment]
@dominikhajduk @php-refactor 
maybe we could run tests using sqlite, not mysql. I mean [this article](https://phpunit.de/manual/current/en/database.html). Storing data sets in xml or yaml files is much easier than in sql format

**Comment by dominikhajduk** [comment]
The key is that database need to be cleared before first test. Problem is that we should use database and selenium extensions. Or run some external script with data fixtures. My solution is the simplest one. Other which I tested wasn't work for me. Because of that I proposed simple one and it can be improve by someone in separate PR. 

Dominik

**Comment by marcinwarzybok** [comment]
dataProvider include "?" instead "ł"

**Comment by partikus** [comment]
@marcinwarzybok do you use vagrant?

**Comment by marcinwarzybok** [comment]
yes, of course

**Comment by partikus** [comment]
@marcinwarzybok I have no idea what could cause a problem, please let us know which line of code does not work.

**Comment by marcinwarzybok** [comment]
Yes, It was of course problem with encoding file. On this computer I got phpstorm without my settings.. School fault ;)
@partikus :+1:

**Comment by marcinwarzybok** [comment]
:+1:
Dnia 22 września 2015 14:41 Dominik Hajduk &lt;notifications@github.com&gt; napisał(a):
Merged #11.—

**Pull request merged**